esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp

```diff
@@ -695,6 +695,31 @@ bool CWsESDLConfigEx::onPublishESDLBinding(IEspContext &context, IEspPublishESDL
     return true;
 }
 
+void setContainerizedEspBindingNameByPort(StringBuffer& espPort, StringBuffer& espBindingName)
+{
+    Owned<IPropertyTree> globalConfig = getGlobalConfigSP();
+    if (!globalConfig)
+        throw makeStringExceptionV(-1, "Can't use global config to find any esp binding for port %s", espPort.str());
+
+    VStringBuffer xpath("services[@class='esp'][@port='%s']", espPort.str());
+    IPropertyTree* process = globalConfig->queryPropTree(xpath);
+    const char* processName = process->queryProp("@name");
```

Comment:
But here you should be making sure you found the process before using the pointer.

Reply:
Also, shouldn't rely on the processName not being the same as the application type.  I think there is a value (either "type" or "application") that you can use instead.

Reply:
fixed both